shared/admin-tools/elasticsearch/get-petition-counts.js

```diff
@@ -0,0 +1,88 @@
+// usage: npx ts-node shared/admin-tools/elasticsearch/get-petition-counts.js 2022
+
+const createApplicationContext = require('../../../web-api/src/applicationContext');
+const {
+  computeDate,
+  dateStringsCompared,
+} = require('../../src/business/utilities/DateHandler');
+const {
+  searchAll,
+} = require('../../src/persistence/elasticsearch/searchClient');
+const { DateTime } = require('luxon');
+
+const year = process.argv[2] || DateTime.now().toObject().year;
+
+const getAllPetitions = async ({ applicationContext }) => {
+  const { results } = await searchAll({
+    applicationContext,
+    searchParameters: {
+      _source: ['createdAt.S', 'isPaper.BOOL'],
+      body: {
+        query: {
+          bool: {
+            must: [
+              {
+                match: {
+                  'eventCode.S': 'P',
+                },
+              },
+              {
+                range: {
+                  'receivedAt.S': {
+                    gte: computeDate({ day: 1, month: 1, year }),
+                    lt: computeDate({ day: 1, month: 1, year: year + 1 }),
+                  },
+                },
+              },
+            ],
+          },
+        },
+        sort: [{ 'receivedAt.S': 'asc' }],
+      },
+      index: 'efcms-docket-entry',
+    },
+  });
+  return results;
+};
+
+const getCounts = ({ gte, lt, petitions }) => {
+  const petitionsReceivedInTimeframe = petitions.filter(
+    p =>
+      dateStringsCompared(p.createdAt, gte) >= 0 &&
+      dateStringsCompared(p.createdAt, lt) < 0,
```

Comment:
Is it possible for paper petitions' `createdAt` and `receivedAt` timestamps to be different?

To retrieve the year's petitions, we're doing a range query on `receivedAt`, but then to filter the petitions by month, we're filtering on `createdAt`. Doing it this way, the total count will still match, but the monthly counts could differ slightly if it's possible that a petition could be received on the last day of the month and entered into DAWSON some days later.